core/lib/vm_interface/src/types/outputs/execution_result.rs

```diff
@@ -95,6 +95,19 @@ impl VmEvent {
             })
             .map(|event| event.indexed_topics[1])
     }
+
+    /// Returns the number of `ContractDeployed` events in the given list of events.
+    pub fn count_contract_deployments(events: &[Self]) -> usize {
+        events
+            .iter()
+            .filter(|event| {
+                event
+                    .indexed_topics
+                    .get(0)
+                    .map_or(false, |&topic| topic == VmEvent::DEPLOY_EVENT_SIGNATURE)
```

Comment:
Shouldn't this check the `address` that emitted the event as well?

Reply:
it definitely should